packages/gestalt/src/utils/wrapWithComponent.js

```diff
@@ -0,0 +1,68 @@
+// @flow strict
+import React, {
+  isValidElement,
+  type ChildrenArray,
+  type ComponentType,
+  type Node,
+} from 'react';
+
+// $FlowIgnore[unclear-type]
+type UnknownComponent = ComponentType<any>;
+
+// In development, we compare based on the name of the function because
+// React Hot Loader proxies React components in order to make updates. In
+// production we can simply compare the components for equality.
+const isComponent = (
+  AComponent: UnknownComponent,
+  AnotherComponent: UnknownComponent
+) => AComponent === AnotherComponent;
+
+// Checks whether `element` is a React element of type `Component` (or one of
+// the passed components, if `Component` is an array of React components).
+function isElementOfType<P>(
+  element: ?Node,
+  Component: ChildrenArray<ComponentType<P>>
+): boolean {
+  if (
+    element === null ||
+    element === undefined ||
+    typeof element !== 'object' // catch all other primitives
+  ) {
+    return false;
+  }
+
+  if (!isValidElement(element) || typeof element.type === 'string') {
+    return false;
+  }
+
+  const { type } = element;
+  const Components = Array.isArray(Component) ? Component : [Component];
+
+  return Components.some(
+    // $FlowFixMe[incompatible-call]
+    (AComponent) => typeof type !== 'string' && isComponent(AComponent, type)
+  );
+}
+
+// Conditionally wraps `element` in `Component` if it is not already an instance of
+// `Component`. If `props` is passed, those will be added as props on the
+// wrapped component. If `element` is nullish, the component is returned unwrapped.
```

Comment:
Is this correct? `If `element` is nullish, the component is returned unwrapped.`, it returns null if so

Reply:
Nope, it's not. I'll put together a lil PR to update. Good catch!